crates/ra_hir/src/resolve.rs

```diff
@@ -183,11 +185,15 @@ impl Resolver {
         Some(res)
     }
 
-    pub(crate) fn resolve_path_in_value_ns(
+    pub(crate) fn resolve_path_in_value_ns<'p>(
         &self,
         db: &impl HirDatabase,
-        path: &Path,
-    ) -> Option<ValueOrPartial> {
+        path: &'p Path,
+    ) -> Option<ResolveValueResult<'p>> {
+        if let PathKind::Type(type_ref) = &path.kind {
+            return Some(ResolveValueResult::TypeRef(type_ref));
+        }
```

Comment:
Hm, this check looks like something that should ideally be done on the call-site, so that we don't have to add this lifetime-bound variant to the `ResolveValueResult`